internal/monitors/conviva/accountservice.go

```diff
@@ -0,0 +1,225 @@
+package conviva
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+)
+
+//const (
+//	metricLensURLFormat = "https://api.conviva.com/insights/2.4/metrics.json?metrics=%s&account=%s&filter_ids=%s&metriclens_dimension_id=%d"
+//)
+
+var anyErrors bool
+var isInitialized bool
+
+// Account for Conviva account data
+type account struct {
+	id                   string
+	Name                 string
+	filters              map[string]string
+	metricLensFilters    map[string]string
+	metricLensDimensions map[string]float64
+}
+
+// AccountService interface for Account related methods
+type accountService interface {
+	getDefault()                                                             *account
+	getMetricLensDimensionMap(accountName string)                            map[string]float64
+	getID(accountName string)                                                string
+	getFilters(accountName string)                                           map[string]string
+	getMetricLensFilters(accountName string)                                 map[string]string
+	getFilterID(accountName string, filterName string)                       string
+	getMetricLensDimensionID(accountName string, metricLensDimension string) float64
+}
+
+type accountServiceImpl struct {
+	defaultAccount *account
+	accounts       []*account
+	ctx            context.Context
+	timeout        *time.Duration
+	client         *httpClient
+	mutex          *sync.RWMutex
+}
+
+// NewAccountService factory function creating AccountService
+func newAccountService(ctx context.Context, timeout *time.Duration, client *httpClient) accountService {
+	service := accountServiceImpl{ctx: ctx, timeout: timeout, client: client, mutex: &sync.RWMutex{},}
+	return &service
+}
+
+func (s *accountServiceImpl) getDefault() *account {
+	s.init()
```

Comment:
This should check init's return value for an error and return early if it errored out.  There should be logging somewhere that an error occurred.